src/lib/util/graph.rs

```diff
@@ -0,0 +1,86 @@
+use petgraph::{
+    self,
+    graph::NodeIndex,
+    visit::{Bfs, IntoNodeReferences, Walker},
+    Direction,
+};
+
+pub struct Graph<T>
```

Comment:
I'm wondering if instead of having a struct that wraps a Graph, we use an extension trait instead that's implemented by `Graph<T, E>`. This way if we have to directly interact with the graph we don't have to do a bunch of graph.inner.whatever stuff. Plus, if one of our *Solve structs adds more fields and can't just be a type synonym for `Graph<T, ()>`, we can directly implement the extension trait on that struct.